### PR TITLE
Update tgocassisrdrgen to export footprint information

### LINKED ISSUE #4473: Add footprint coordinates for TGO CaSSIS images to exported label
**Description**  
Update `tgocassisrdrgen` to output footprint coordinates which will be added by the UA team to the ISIS Cube labels of CaSSIS images.

**Example**  

Take the following keywords/values from the `Archive` group of a CaSSIS cube: 

```
Corner1Longitude = <lon1>
Corner1Latitude = <lat1>
.
.
.
Corner4Longitude = <lon4>
Corner4Latitude = <lat4>
```

and update `tgocassisrdrgen` to export these values to the PDS4 label it creates. 

(Example of desired output forthcoming)

### PR DESCRIPTION
**Pull request by krlberry**
## Description
Updates `tgocassisrdrgen` to export footprint information added by the geometry pipeline to the cube label as described in ticket #4473 

## Related Issue
Closes #4473

## Motivation and Context
Updates the exported PDS4 product to contain footprint information that was added to the cube label by the geometry pipeline. 

## How Has This Been Tested?
Tested by-hand on constructed labels.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/tgo/apps/tgocassisrdrgen/main.cpp [hunk 1]
```diff
@@ -191,6 +191,31 @@ void IsisMain() {
...
+  QDomElement pixelInterceptNode4 = pixelInterceptNode3.nextSiblingElement();
+
+  pixelInterceptNode.insertAfter(pixelInterceptNode.firstChildElement("geom:pixel_latitude"), pixelInterceptNode.firstChildElement("geom:reference_pixel_location"));
+  pixelInterceptNode2.insertAfter(pixelInterceptNode2.firstChildElement("geom:pixel_latitude"), pixelInterceptNode2.firstChildElement("geom:reference_pixel_location"));
+  pixelInterceptNode3.insertAfter(pixelInterceptNode3.firstChildElement("geom:pixel_latitude"), pixelInterceptNode3.firstChildElement("geom:reference_pixel_location"));
+  pixelInterceptNode4.insertAfter(pixelInterceptNode4.firstChildElement("geom:pixel_latitude"), pixelInterceptNode4.firstChildElement("geom:reference_pixel_location"));
+  
+  pixelInterceptNode2.setTagName("geom:Pixel_Intercept");
+  pixelInterceptNode3.setTagName("geom:Pixel_Intercept");
+  pixelInterceptNode4.setTagName("geom:Pixel_Intercept");
```
Comment (jessemapel) [inline isis/src/tgo/apps/tgocassisrdrgen/main.cpp]: Why is this needed? I'd expect the translation file to put these in the correct places based on the siblings.

Comment (krlberry) [reply isis/src/tgo/apps/tgocassisrdrgen/main.cpp]: @jessemapel This is a work-around because I couldn't get it working just via the translation files. I know you worked on the original code for ProcessExportPds4 (etc) as well, do you know if there's a better way? I might have forgotten the right way to handle this.

When I just had:
(The difference is that I removed the numbers from the end of the `geom:Pixel_Intercept` values and added the `OutputSiblings` entry to the longitude translation groups in addition to the latitude translation groups.)
```
Group = Corner1Lattiude
  Auto
  Optional
  InputKey        = Corner1Latitude
  InputPosition   = (IsisCube, Archive)
  OutputSiblings =  ("geom:reference_pixel_location|Lower Left Corner")
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  Translation     = (*, *)
End_Group

Group = Corner1Longitude
  Auto
  Optional
  InputKey        = Corner1Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  Translation     = (*, *)
End_Group

Group = Corner2Lattiude
  Auto
  Optional
  InputKey        = Corner2Latitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, new@geom:Pixel_Intercept)
  OutputSiblings = ("geom:reference_pixel_location|Lower Right Corner")
  Translation     = (*, *)
End_Group

Group = Corner2Longitude
  Auto
  Optional
  InputKey        = Corner2Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  OutputSiblings = ("geom:reference_pixel_location|Lower Right Corner")
  Translation     = (*, *)
End_Group

Group = Corner3Lattiude
  Auto
  Optional
  InputKey        = Corner3Latitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, new@geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Left Corner")
  Translation     = (*, *)
End_Group

Group = Corner3Longitude
  Auto
  Optional
  InputKey        = Corner3Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Left Corner")
  Translation     = (*, *)
End_Group

Group = Corner4Lattiude
  Auto
  Optional
  InputKey        = Corner4Latitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_latitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, new@geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Right Corner")
  Translation     = (*, *)
End_Group

Group = Corner4Longitude
  Auto
  Optional
  InputKey        = Corner4Longitude
  InputPosition   = (IsisCube, Archive)
  OutputName      = geom:pixel_longitude
  OutputPosition  = (Product_Observational, Observation_Area, Discipline_Area, geom:Geometry, geom:Geometry_Orbiter, geom:Surface_Geometry, geom:Surface_Geometry_Specific, geom:Footprint_Vertices, geom:Pixel_Intercept)
  OutputSiblings  = ("geom:reference_pixel_location|Upper Right Corner")
  Translation     = (*, *)
End_Group
``` 

I got the following: 

.
.
.
```
 <geom:Geometry>
    <geom:Geometry_Orbiter>
     <geom:Surface_Geometry>
      <geom:Surface_Geometry_Specific>
       <geom:Footprint_Vertices>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Lower Left Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">10</geom:pixel_latitude>
         <geom:pixel_longitude unit="deg">1</geom:pixel_longitude>
         <geom:pixel_longitude unit="deg">2</geom:pixel_longitude>
         <geom:pixel_longitude unit="deg">3</geom:pixel_longitude>
         <geom:pixel_longitude unit="deg">4</geom:pixel_longitude>
        </geom:Pixel_Intercept>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Lower Right Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">12</geom:pixel_latitude>
        </geom:Pixel_Intercept>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Upper Left Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">13</geom:pixel_latitude>
        </geom:Pixel_Intercept>
        <geom:Pixel_Intercept>
         <geom:reference_pixel_location>Upper Right Corner</geom:reference_pixel_location>
         <geom:pixel_latitude unit="deg">14</geom:pixel_latitude>
        </geom:Pixel_Intercept>
       </geom:Footprint_Vertices>
      </geom:Surface_Geometry_Specific>
     </geom:Surface_Geometry>
    </geom:Geometry_Orbiter>
   </geom:Geometry>
```

Comment (jessemapel) [reply isis/src/tgo/apps/tgocassisrdrgen/main.cpp]: Strange, I don't know why that's happening. It looks like it should spit things out in the expected places. Considering the isisexport work being done, I think this is okay.

**Review by jessemapel — DISMISSED** [review]
Are there expected changes in test outputs for this?

**Comment by krlberry** [comment]
@jessemapel Yes, I updated the input and truthdata for one category test.

**Comment by Kelvinrr** [comment]
Seems like the TGO module tests are failing with comparison tests? I would assume there needs more changes to other cat tests?

**Comment by krlberry** [comment]
@Kelvinrr This is now fixed. It was a real issue, thanks for pointing it out and sorry I didn't see your comment yesterday.

**Review by jessemapel — APPROVED** [review]
